receiver/prometheusreceiver/internal/prom_to_otlp_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"sort"
+	"testing"
+	"unsafe"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/testing/protocmp"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	otlpresource "go.opentelemetry.io/collector/internal/data/protogen/resource/v1"
+	"go.opentelemetry.io/collector/translator/internaldata"
+)
+
+func TestCreateNodeAndResourceConversion(t *testing.T) {
+	job, instance, scheme := "converter", "ocmetrics", "http"
+	ocNode, ocResource := createNodeAndResource(job, instance, scheme)
+	mdFromOC := internaldata.OCToMetrics(internaldata.MetricsData{
```

Comment:
Well I am doing a straight up translation of what exists in the code and I need to produce the exact same result that the current code produces. This PR converts from Prometheus->OTLP instead of Prometheus->OpenCensus->OTLP, and once the entire translation is done and we are sure that we have an absolute 1-to-1 mapping, these tests will be discarded.

Reply:
Perhaps the test can be renamed in a way that makes it clear it is only intended to validate congruence with the outgoing implementation and will need to be removed.

Reply:
Sounds good @Aneurysm9, I've renamed it to `TestCreateNodeAndResourceEquivalence`